### PR TITLE
Website config update

### PR DESCRIPTION
**Pull request by vbuberen**
## Description

Initially I planned to just update the workflow on Github Actions for website deployment, but saw a few other minor things to fix, like wrong favicon used (looks like from Flutterfire), so did more changes:

- Updated Github Actions workflow to use fresh and stable versions of 3rd party actions.
- Bumped Node version to 16 as Docusaurus supports it
- Replaced favicon with the one from Flutter Community
- Bumped a few packages versions for npm (those which were non-breaking)
- Modified supported browsers list to use [default target list](https://github.com/browserslist/browserslist#best-practices)
- Removed some redundant images not used by the website

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated the version in `pubspec.yaml` and `CHANGELOG.md`.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

File: website/package.json [hunk 1]
```diff
@@ -14,33 +14,23 @@
...
-  },
+  "browserslist": [
+    "defaults"
+  ],
   "dependencies": {
     "@docusaurus/core": "^2.0.0-beta.0",
     "@docusaurus/module-type-aliases": "^2.0.0-beta.0",
     "@docusaurus/plugin-google-gtag": "^2.0.0-beta.0",
     "@docusaurus/preset-classic": "^2.0.0-beta.0",
-    "@types/classnames": "^2.3.1",
```
Comment (vbuberen) [inline website/package.json:37]: Removed, since it is not needed anymore https://www.npmjs.com/package/@types/classnames

**Comment by vbuberen** [comment]
Looks like all_plugins workflow has some configuration issues if started checks for such PR.

**Comment by vbuberen** [comment]
Requested review from people suggested by Github. Not sure if it was a correct choice, so let me know if I should request a review from someone else.

While working on this PR I saw a few points to improve, which I would like us to implement as well in nearest future:
- Update Docusaurus to more recent version
- Add preview channel to have a preview available in PRs before merging the PR with website modifications. It can be done multiple ways, but one of options is to use [the official Firebase Action ](https://github.com/marketplace/actions/deploy-to-firebase-hosting). I don't know who has access to Firebase setup, so would appreciate if somebody shares info about it.
- Move from `npm` to `yarn` (might be a little bit faster to deploy in future)

**Review by mhadaily — APPROVED** [review]
LGTM, 

your feature plan looks also good expect I don't think we will benefit much from Yarn but you'll figure out when we have next PR> 

thank you

**Pull request merged**